src/extractor/guidance/node_based_graph_walker.cpp

```diff
@@ -167,6 +168,37 @@ operator()(const NodeID /*nid*/,
          angularDeviation(min_element->angle, STRAIGHT_ANGLE) < NARROW_TURN_ANGLE) &&
         angularDeviation(initial_bearing, min_element->bearing) < NARROW_TURN_ANGLE;
 
+    // do not allow major turns in the road, if other similar turns are present
+    // e.g.a turn at the end of the road:
+    //
+    // a - - a - - a - b - b
+    //             |
+    //       a - - a
+    //             |
+    //             c
+    //
+    // Such a turn can never be part of a merge
+    // We check if there is a similar turn to the other side. If such a turn exists, we consider the
+    // continuation of the road not possible
+    if (util::angularDeviation(STRAIGHT_ANGLE, min_element->angle) > GROUP_ANGLE)
+    {
+        auto deviation = util::angularDeviation(STRAIGHT_ANGLE, min_element->angle);
+        auto opposite_angle = min_element->angle >= STRAIGHT_ANGLE ? (STRAIGHT_ANGLE - deviation)
```

Comment:
here the opposite angle is a mirrored one. Let's look at https://www.openstreetmap.org/node/1457716777
![figure_1](https://user-images.githubusercontent.com/4421046/31810963-0b59c82e-b57e-11e7-88b2-ec460c2fe7df.png)
where the dashed road is https://www.openstreetmap.org/way/132530815, 132.5 degrees turn to https://www.openstreetmap.org/way/132531140 and 300.8 degrees turn to https://www.openstreetmap.org/way/132530814

`min_element` is the road with 300.8 degrees angle, `deviation` is 120 degrees, and `opposite_angle` is 180-120=60 degrees that is is a green line and the closest is the U-turn road with 0 degrees rather than the turn right with 132.5 degrees. Non-mirrored opposite angle for 300 degrees will be (300+180)%360=120 degrees or the red line.